ostd/src/boot/smp.rs

```diff
@@ -124,8 +124,14 @@ pub fn register_ap_entry(entry: fn()) {
     AP_LATE_ENTRY.call_once(|| entry);
 }
 
+/// Early boot entry of an AP.
+///
+/// # Safety
+///
+/// - This function must be called only once on each AP.
+/// - The caller must follow C calling conventions and put the right arguments in registers.
 #[no_mangle]
-fn ap_early_entry(cpu_id: u32) -> ! {
+pub(crate) unsafe extern "C" fn ap_early_entry(cpu_id: u32) -> ! {
```

Comment:
NOTE: I changed three things here
* make it unsafe as the intention of this PR.
* make it `extern "C"`, and attach a comment in ostd/src/arch/x86/boot/ap_boot.S to ensure cpu_id is passed as per C calling conventions.
* make it visible in crate to let riscv_ap_early_entry replace the import of ap_early_entry symbol by module-based path.